### PR TITLE
Feature/message buttons

### PR DESCRIPTION
**Pull request by lautimfernandez**
## Motivation
Adding a new feature for message component.

## Description
Please describe your intentions with this new code you want to introduce inside Andes UI lib.
Try to answer those questions:
- What is it about?
+ Adding a new feature to the message
- How does it work?
+ Allowing to have primary and secondary actions

## Affected component
Andes Message

## Frontify component link
https://company-161429.frontify.com/d/kxHCRixezmfK/n-a#/componentes/message

## Screenshots
Screenshot_1580504642

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [X] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [X] I have modified the [Changelog](https://github.com/mercadolibre/...) file,
- [ ] I have updated GitHub wiki,
- [X] I have the explicit approval of one or more members of the UX Team,
- [X] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [X] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/src/main/java/com/mercadolibre/android/andesui/button/AndesButton.kt [hunk 1]
```diff
@@ -295,4 +297,12 @@ class AndesButton : ConstraintLayout {
         leftIconComponent.isEnabled = enabled
         rightIconComponent.isEnabled = enabled
     }
+
+    internal fun changeTextColor(color : Int){
+        textComponent.setTextColor(color)
+    }
+
+    fun changeBackgroundColor(backgroundColorConfig: BackgroundColorConfigMessage ){
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/button/AndesButton.kt]:
```suggestion
    fun changeBackgroundColor(backgroundColorConfig: BackgroundColorConfigMessage){
```

File: components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt [hunk 2]
```diff
@@ -68,13 +72,28 @@ class AndesMessage : FrameLayout {
             setupDismissable(createConfig())
         }
 
+    private var primatyActionText: String
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt]: typo: primaTy

File: components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt [hunk 3]
```diff
@@ -217,6 +239,42 @@ class AndesMessage : FrameLayout {
...
+        primaryAction.changeBackgroundColor(config.primaryActionBackgroundColor)
+        primaryAction.changeTextColor(config.primaryActionTextColor)
+        secondaryAction.changeBackgroundColor(config.secondaryActionBackgroundColor)
+        secondaryAction.changeTextColor(config.secondaryActionTextColor)
+    }
+
+    fun setupPrimaryAction(text: String, onClickListener: OnClickListener) {
+        primaryAction.visibility = View.VISIBLE
+        primatyActionText = text
+        primaryAction.setOnClickListener(onClickListener)
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt]: Qué pasa si el `onClickListener` es null?

File: components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt [hunk 4]
```diff
@@ -217,6 +239,42 @@ class AndesMessage : FrameLayout {
...
+    fun hidePrimaryAction(){
+        primaryAction.visibility = View.GONE
+        secondaryAction.visibility = View.GONE
+    }
+
+    fun hideSecondaryAction(){
+        secondaryAction.visibility = View.GONE
+
+    }
+
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt]: empty line

File: components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt [hunk 5]
```diff
@@ -217,6 +239,42 @@ class AndesMessage : FrameLayout {
...
+    fun hidePrimaryAction(){
+        primaryAction.visibility = View.GONE
+        secondaryAction.visibility = View.GONE
+    }
+
+    fun hideSecondaryAction(){
+        secondaryAction.visibility = View.GONE
+
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/message/AndesMessage.kt]: empty line

File: components/src/main/java/com/mercadolibre/android/andesui/message/factory/AndesMessageConfigurationFactory.kt [hunk 6]
```diff
@@ -66,4 +74,9 @@ internal object AndesMessageConfigurationFactory {
     private fun resolveIcon(type: AndesMessageTypeInterface, hierarchy: AndesMessageHierarchyInterface, context: Context) = type.icon(context, hierarchy)
     private fun resolveDismissableIcon(hierarchy: AndesMessageHierarchyInterface, context: Context) = hierarchy.dismissableIcon(hierarchy, context)
     private fun resolveDismissableIconColor(hierarchy: AndesMessageHierarchyInterface, context: Context) = hierarchy.dismissableIconColor(context)
+    private fun resolvePrimaryActionBackgroundColor(hierarchy: AndesMessageHierarchyInterface, state: AndesMessageTypeInterface, context: Context) = hierarchy.primaryActionBackgroundColor(context, state)
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/message/factory/AndesMessageConfigurationFactory.kt]: Debería ser `type` en vez de `state`

File: components/src/main/java/com/mercadolibre/android/andesui/message/factory/AndesMessageConfigurationFactory.kt [hunk 7]
```diff
@@ -66,4 +74,9 @@ internal object AndesMessageConfigurationFactory {
...
+    private fun resolvePrimaryActionTextColor(hierarchy: AndesMessageHierarchyInterface, context: Context) = hierarchy.primaryActionTextColor(context)
+    private fun resolveSecondaryActionBackgroundColor(hierarchy: AndesMessageHierarchyInterface, state: AndesMessageTypeInterface, context: Context) = hierarchy.secondaryActionBackgroundColor(context, state)
+    private fun resolveSecondaryActionTextColor(hierarchy: AndesMessageHierarchyInterface, state: AndesMessageTypeInterface, context: Context) = hierarchy.secondaryActionTextColor(context, state)
+
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/message/factory/AndesMessageConfigurationFactory.kt]: empty line

File: components/src/main/java/com/mercadolibre/android/andesui/message/hierarchy/AndesMessageHierarchyInterface.kt [hunk 8]
```diff
@@ -83,18 +86,31 @@ internal sealed class AndesMessageHierarchyInterface {
...
+            iconBackgroundColor(context, type)!!,iconBackgroundColor(context, type)!!, iconBackgroundColor(context, type)!!, iconBackgroundColor(context, type)!!)
+
+    fun primaryActionTextColor(context: Context) = ContextCompat.getColor(context, R.color.andesui_white)
+
+    fun secondaryActionBackgroundColor(context: Context, type: AndesMessageTypeInterface) = BackgroundColorConfigMessage(ContextCompat.getColor(context, R.color.andesui_bu_transparent_idle), ColorUtils.blendARGB(ContextCompat.getColor(context, R.color.andesui_bu_transparent_idle), Color.BLACK, 0.2f), backgroundColor(context, type),
+            backgroundColor(context, type), backgroundColor(context, type), backgroundColor(context, type))
+
+    abstract fun secondaryActionTextColor(context: Context, type: AndesMessageTypeInterface) : Int
+
+
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/message/hierarchy/AndesMessageHierarchyInterface.kt]: empty lines

**Pull request merged**